java/src/main/java/io/grpc/examples/wallet/AccountServer.java

```diff
@@ -117,20 +118,37 @@ private void start() throws IOException {
     // Since the main server may be using TLS, we start a second server just for plaintext health
     // checks
     int healthPort = port + 1;
-    server =
-        XdsServerBuilder.forPort(port, serverCredentials)
-            .addService(
-                ServerInterceptors.intercept(
-                    new AccountImpl(), new WalletInterceptors.HostnameInterceptor()))
-            .addService(ProtoReflectionService.newInstance())
-            .addService(health.getHealthService())
-            .build()
-            .start();
-    healthServer =
-        XdsServerBuilder.forPort(healthPort, InsecureServerCredentials.create())
-            .addService(health.getHealthService()) // allow management servers to monitor health
-            .build()
-            .start();
+    if (credentialsType == CredentialsType.XDS) {
+      server =
+          XdsServerBuilder.forPort(port, serverCredentials)
+              .addService(
+                  ServerInterceptors.intercept(
+                      new AccountImpl(), new WalletInterceptors.HostnameInterceptor()))
+              .addService(ProtoReflectionService.newInstance())
+              .addService(health.getHealthService())
+              .build()
+              .start();
+      healthServer =
+          XdsServerBuilder.forPort(healthPort, InsecureServerCredentials.create())
+              .addService(health.getHealthService()) // allow management servers to monitor health
+              .build()
+              .start();
```

Comment:
The `healthServer` will always be insecure. Let's move it out.

Reply:
As discussed offline there is a reason to build health service with `XdsServerBuilder` even if always insecure